javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,98 @@
+let userName = null; 
+let todoList = []; 
+
+function getReply(command) {
+  const lowerCaseCommand = command.toLowerCase();
+
+  if (lowerCaseCommand.startsWith('hello my name is ')) {
+    const name = command.slice(17); 
+    if (userName === name) {
+      return `You already introduced yourself as ${name}.`;
+    } else {
+      userName = name;
+      return `Nice to meet you ${name}`;
+    }
+  }
+
+  if (lowerCaseCommand === 'what is my name?') {
+    return userName
+      ? `Your name is ${userName}`
+      : "I don't know your name yet. Please tell me your name first.";
+  }
+
+  if (
+    lowerCaseCommand.startsWith('add ') &&
+    lowerCaseCommand.includes(' to my todo')
+  ) {
+    const todo = command.slice(4, command.indexOf(' to my todo'));
```

Comment:
It's not something super important to handle, but it's always good to think about edge cases and how our code might behave.

For example, what if someone wanted to add `think of some things to add to my todo` to their todo list? Their command would be:
```
add think of some things to add to my todo to my todo
```
Which is a little silly for sure! But it would be nice if our code handled it.

What the code will currently do is see the first instance of `to my todo` when we call `command.indexOf("to my todo")`. That means it will assign `think of some things to add` to the `todo` variable, whereas it would ideally assign `think of some things to add to my todo` instead.

We can fix this by using a slightly different method from `indexOf()`. Fortunately Javascript has a `lastIndexOf()` method for exactly this reason. Instead of returning the index of the first matching string, it will return the index of the last one!

Again - not a super important fix for now. The lesson is mostly just to keep thinking about our code and how it might break, so that we can make deliberate choices about whether we fix it!